### DISCUSSION TITLE
Instagram: profile picture max size?

### DISCUSSION THREAD

**Discussion by DevWannabe-dot** [qsniyg/maxurl#909] [Ideas]
I've been through lots of sites and tried lots of different tools to get the biggest image size for any profile picture on Instagram, obtaining no success. First of all you got to use Absolute Enable Right Click & Copy, either the extension or the userscript to open the protected image in new tab. The problem is that facebook-hosted photos aren't resizable or have a regular filename. I've tried breaking down the URLs I got from specific online services like instadp.net, example:

## Resized

https://instagram.fplu3-1.fna.fbcdn.net/v/t51.2885-19/s150x150/242156895_388316952797513_8957624082015401668_n.jpg?_nc_ht=instagram.fplu3-1.fna.fbcdn.net&_nc_ohc=o1R_2MqHYQoAX9w2d08&edm=ABfd0MgBAAAA&ccb=7-4&oh=b233cd4cfe5ca8b8e201c7845ce8ecef&oe=61510FE7&_nc_sid=7bff83

> https://instagram.fplu3-1.fna.fbcdn.net/v/t51.2885-19/s150x150/242156895_388316952797513_8957624082015401668_n.jpg

> ?_nc_ht=instagram.fplu3-1.fna.fbcdn.net&_nc_ohc=o1R_2MqHYQoAX9w2d08&edm=ABfd0MgBAAAA&ccb=7-4&oh=b233cd4cfe5ca8b8e201c7845ce8ecef&oe=61510FE7&_nc_sid=7bff83

## Original

https://scontent-amt2-1.cdninstagram.com/v/t51.2885-19/s320x320/242156895_388316952797513_8957624082015401668_n.jpg?_nc_ht=scontent-amt2-1.cdninstagram.com&_nc_ohc=o1R_2MqHYQoAX_2m9q8&tn=mRhom0o06UqAqeJv&edm=ABfd0MgBAAAA&ccb=7-4&oh=d57963f92dfb85d8098ceeaa32394896&oe=6150771F&_nc_sid=7bff83 

> https://scontent-amt2-1.cdninstagram.com/v/t51.2885-19/s320x320/242156895_388316952797513_8957624082015401668_n.jpg

> ?_nc_ht=scontent-amt2-1.cdninstagram.com&_nc_ohc=o1R_2MqHYQoAX_2m9q8&tn=mRhom0o06UqAqeJv&edm=ABfd0MgBAAAA&ccb=7-4&oh=d57963f92dfb85d8098ceeaa32394896&oe=6150771F&_nc_sid=7bff83

But even after comparing these, I notice they are different domains and some of the string presented simply can't be compared like I did on #833. Then I find about this [article](https://stackoverflow.com/questions/50086945/how-to-get-instagram-profile-picture-via-api), that says we can get some basic data as json by Instagram api on the URL:
https://www.instagram.com/USERNAME/?__a=1

Testing with https://www.instagram.com/teddysphotos/?__a=1 , this is what I've found (minimized URLs, a pretty-printed json page on firefox)
![image](https://user-images.githubusercontent.com/64389041/134325640-d696c804-25d4-4cd7-aacd-a191cdb69a5c.png)

If I compare these two URLs, fairly more similar (had to remove https://  for formatting), still I can't do much with the bolded strings:

## Resized

instagram.fplu3-1.fna.fbcdn.net/v/t51.2885-19/**s150x150**/238516208_3231477057079510_7540080569771000360_n.jpg?_nc_ht=instagram.fplu3-1.fna.fbcdn.net&_nc_ohc=iJkCq9GBvpIAX9RUIT_&edm=ABfd0MgBAAAA&ccb=7-4&oh=**20777e01cc0d76a1bd5b6dc99b90772c**&oe=**615103EC**&_nc_sid=7bff83

## Original 

instagram.fplu3-1.fna.fbcdn.net/v/t51.2885-19/**s320x320**/238516208_3231477057079510_7540080569771000360_n.jpg?_nc_ht=instagram.fplu3-1.fna.fbcdn.net&_nc_ohc=iJkCq9GBvpIAX9RUIT_&edm=ABfd0MgBAAAA&ccb=7-4&oh=**37ecaab313d0faa8d9f0211e9f6bac66**&oe=**6151FDFD**&_nc_sid=7bff83

### Bottom line

So, considering maxurl consists of hardcoded websites, as a functional single-file js userscript, would it even be possible to, maybe with the popup, make a request to https://www.instagram.com/USERNAME/?__a=1 by asking the user to input only the username when in a http://instagram.fplu3-1.fna.fbcdn.net page and then display _profile_pic_url_hd_ on the popup frame? I mean, there is no way we can tell what instagram profile the profile picture belongs to if we only have the url. It's even harder to find out what are `&oh=` and `&oe=`, as it would probably require another workaround as well.

And all of this is because it's an advantage no other software tool besides these advertised blogs you can find with the promise to give you the biggest profile picture, but are no practical. I'll give any extra info I can if asked, and hope all of this makes sense.

**Comment by qsniyg** [comment]
Good idea about supporting `?__a=1` for the non-native API, but for the moment, you can also enable `Instagram: Use native API` to get a 1080x* image :)

I should note that the `oh` key refers to the hash/signature (`oe` I believe is the expiry time), and it hashes most parts of the URL, so it's not possible to modify it much unfortunately. We can remove a few parameters like `se` (which helps image quality: #239), but we can't for example change the picture resolution

**Comment by DevWannabe-dot** [comment]
> you can also enable Instagram: Use native API to get a 1080x* image :)

> We can remove a few parameters like se (which helps image quality

Didn't know about these, thanks! So, do you think there is some way this can be done or only with other extensions? By the way, `Instagram: Use native API` stands for what?

**Reply by qsniyg** [reply]
If you're referring to removing &se=, IMU does this already :)

`Instagram: Use native API` is a setting under the Rules tab that uses Instagram's mobile app API, rather than their web API. This allows for higher resolution images, such as 1440x\* for posts and 1080x\* for profile pictures. However, it's increasingly dangerous to use it with a real account, so that's why it's disabled for now. Hopefully someday they will provide the larger images to their web interface so ugly workarounds like this aren't necessary!

**Reply by DevWannabe-dot** [reply]
> 
> 
> If you're referring to removing &se=, IMU does this already :)
> 
> `Instagram: Use native API` is a setting under the Rules tab that uses Instagram's mobile app API, rather than their web API. This allows for higher resolution images, such as 1440x* for posts and 1080x* for profile pictures. However, it's increasingly dangerous to use it with a real account, so that's why it's disabled for now. Hopefully someday they will provide the larger images to their web interface so ugly workarounds like this aren't necessary!

Alright, I get it. I've got a backup account on a different IP and I wanna see how that works, as I've seen not enough resources in the web. Can you help me? I've seen no difference opening the images in new tab or in a post itself, but you said it's disabled, so what do I do? Thanks.
